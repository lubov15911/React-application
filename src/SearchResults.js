import React, { Component } from 'react';

import Sum from './TestComponent'; // Component for test

export default class SearchResults extends Component {
    render() {
        return (
            <div>
              <p>{Sum(11, 6)} movies found</p>
              <p>Sort by</p>
            </div>
        );
    }
}
