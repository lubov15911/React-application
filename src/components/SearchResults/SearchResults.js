import React from 'react';

import EmptyList from "../EmptyList/EmptyList";
import ResultsList from "../ResultsList/ResultsList";

import './SearchResults.scss';

const SearchResults = (props) => {
    return (
        <div className={`search-results ${!props.totalResults && 'no-items'}`}>
            {props.totalResults && <ResultsList films={props.films} raiseClickEvent={props.raiseClickEvent} />}
            {!props.totalResults && <EmptyList />}
        </div>
    );
};
export default SearchResults;
