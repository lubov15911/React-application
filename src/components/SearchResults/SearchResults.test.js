import React from 'react';
import { shallow } from 'enzyme';

import SearchResults from './SearchResults';

describe('SearchResults', () => {
    it('should render correctly films array', () => {
        const fakeFunction = () => {};
        const films = [{
            id: 354912,
            poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
            title: 'Coco',
            release_date: '2017-10-27',
            genres: ["Adventure", "Comedy", "Family", "Animation"],
        }, {
            id: 333339,
            poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
            title: 'Ready Player One',
            release_date: '2018-03-28',
            genres: ["Adventure", "Science Fiction", "Action"],
        }] ;

        const component = shallow(<SearchResults
            totalResults={films.length}
            films={films}
            raiseClickEvent={fakeFunction}
        />);

        expect(component).toMatchSnapshot();
    });

    it('should render correctly empty films array', () => {
        const fakeFunction = () => {};
        const films = [] ;

        const component = shallow(<SearchResults
            totalResults={films.length}
            films={films}
            raiseClickEvent={fakeFunction}
        />);

        expect(component).toMatchSnapshot();
    });
});
