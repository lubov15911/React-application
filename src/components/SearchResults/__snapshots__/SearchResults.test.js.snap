// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render correctly empty films array 1`] = `
<div
  className="search-results no-items"
>
  0
  <EmptyList />
</div>
`;

exports[`SearchResults should render correctly films array 1`] = `
<div
  className="search-results false"
>
  <ResultsList
    films={
      Array [
        Object {
          "genres": Array [
            "Adventure",
            "Comedy",
            "Family",
            "Animation",
          ],
          "id": 354912,
          "poster_path": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
          "release_date": "2017-10-27",
          "title": "Coco",
        },
        Object {
          "genres": Array [
            "Adventure",
            "Science Fiction",
            "Action",
          ],
          "id": 333339,
          "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
          "release_date": "2018-03-28",
          "title": "Ready Player One",
        },
      ]
    }
    raiseClickEvent={[Function]}
  />
</div>
`;
